src/agent/discovery_proxy.cpp

```diff
@@ -214,25 +175,52 @@ void DiscoveryProxy::OnServiceDiscovered(const std::string &
     instanceInfo.mTxtData   = aInstanceInfo.mTxtData.data();
     instanceInfo.mTtl       = CapTtl(aInstanceInfo.mTtl);
 
-    std::for_each(mSubscriptions.begin(), mSubscriptions.end(), [&](const MdnsSubscription &aSubscription) {
-        if (aSubscription.MatchesServiceInstance(aType, aInstanceInfo.mName))
+    while ((query = otDnssdGetNextQuery(mNcp.GetInstance(), query)) != nullptr)
+    {
+        std::string      instanceName, serviceName, hostName, domain;
+        char             queryName[OT_DNS_MAX_NAME_SIZE];
+        otDnssdQueryType type = otDnssdGetQueryTypeAndName(query, &queryName);
+        otbrError        splitError;
+
+        switch (type)
+        {
+        case OT_DNSSD_QUERY_TYPE_BROWSE:
+            splitError = SplitFullServiceName(queryName, serviceName, domain);
+            break;
+        case OT_DNSSD_QUERY_TYPE_RESOLVE:
+            splitError = SplitFullServiceInstanceName(queryName, instanceName, serviceName, domain);
+            break;
+        case OT_DNSSD_QUERY_TYPE_RESOLVE_HOST:
+            splitError = OTBR_ERROR_PARSE;
+            break;
+        default:
+            splitError = OTBR_ERROR_NOT_IMPLEMENTED;
+            break;
```

Comment:
```suggestion
        default:
            splitError = OTBR_ERROR_NOT_FOUND;
            break;
```
seems we don't need to differentiate `OT_DNSSD_QUERY_TYPE_RESOLVE_HOST` and default?